source/adapters/level_zero/memory.hpp

```diff
@@ -158,6 +158,14 @@ struct _ur_buffer final : ur_mem_handle_t_ {
   // Tells the host allocation to use for buffer map operations.
   char *MapHostPtr{nullptr};
 
+  // Flag to indicate if this is a device duffer created as a proxy for a native
+  // buffer handle.
+  bool DeviceMappedNativeBuffer{false};
```

Comment:
Wouldn't checking `DeviceMappedHostNativePtr != null` be enough?

Reply:
I realized that I had originally created the 2 trackers since I thought there was more than one condition, but yes there is only one object to check, I have fixed this in the latest version, thanks!